src/renderers/dom/shared/flattenStyleArray.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule flattenStyleArray
+ * @flow
+ */
+
+'use strict';
+
+type NestedStylesArray = Array<Object | NestedStylesArray | null>;
```

Comment:
We should also allow false as it's nice to be able to do `cond && style` which will return `false` when cond is `false`